### PR TITLE
Update serialization docs, #222

### PR DESCRIPTION
**Pull request by patriknw**
* add samples/doc-snippets where we can add small doc includes
  when a full example isn't motivated
* documented the JsonSupport

Refs #222

### CODE REVIEW

**Comment by patriknw** [comment]
It didn't really work though. I created issue https://github.com/lightbend/akkaserverless-java-sdk/issues/324

File: samples/doc-snippets/src/main/java/com/example/json/MyServiceAction.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    LOG.info("Consumed " + jsonMessage);
+    return effects().reply(Empty.getDefaultInstance());
+  }
+
+  @Override
+  public Effect<Any> produce(MyServiceApi.KeyValue keyValue) {
+    JsonKeyValueMessage jsonMessage =
+        new JsonKeyValueMessage(keyValue.getKey(), keyValue.getValue()); // <2>
+    Any jsonAny = JsonSupport.encodeJson(jsonMessage); // <3>
+    return effects().reply(jsonAny);
```
Comment (johanandren) [inline samples/doc-snippets/src/main/java/com/example/json/MyServiceAction.java:35]: I don't quite get the API here, is this a reply to whoever called the action, or is it the message to publish or both?

[the remark this answers is not in the document]
Comment (patriknw) [reply samples/doc-snippets/src/main/java/com/example/json/MyServiceAction.java:35]: Normally, publishing would be from an eventing.in from journal and then the reply is only an ack (not visible, no client caller)

**Review by octonato — APPROVED** [review]

File: samples/doc-snippets/src/main/java/com/example/json/MyServiceAction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (patriknw) [reply samples/doc-snippets/src/main/java/com/example/json/MyServiceAction.java:35]: The alternative that was discussed when adding the publishing support was to have an explicit context.publish (now that would be an effect), but there were reasons (by James) to go with the return value and proto annotations for `eventing.out`.

[the remark this answers is not in the document]
Comment (johanandren) [reply samples/doc-snippets/src/main/java/com/example/json/MyServiceAction.java:35]: Ok, so now with the correct encoding for the json-in-any, that still fails in the same way, while the topic publish/consume works without problems. That should mean that publishing through an action only ever worked with regular protobuf data types and not the akkaserverless-json. (Maybe that is fine?)

**Pull request merged**